torch/_ops.py

```diff
@@ -925,6 +935,91 @@ def tags(self):
     # TODO: add more methods to expose information about input and output arguments
 
 
+class CustomOpOverload(OpOverload):
+    def __repr__(self):
+        return "<CustomOpOverload(op='{}.{}', overload='{}')>".format(
+            *self._schema.name.split("::"), self._overloadname
+        )
+
+    def _fallthrough_keys(self) -> list[DispatchKey]:
+        # TODO: we should be calling the fallback for these, but a fallthrough is almost close
+        # enough to the fallback in most cases that we care about.
+        _DEFAULT_FALLTHROUGH_KEYS = [
+            DispatchKey.Autograd,
+            DispatchKey.AutogradCPU,
+            DispatchKey.AutogradCUDA,
+            DispatchKey.ADInplaceOrView,
+            DispatchKey.BackendSelect,
+            DispatchKey.PythonTLSSnapshot,
+            DispatchKey.PythonDispatcher,
+        ]
```

Comment:
BUG: In eager-mode, we never actually hit AutogradCPU. Somehow the fallthrough takes higher priority than the registered kernel. The fallthrough should not take higher priority.

In the ideal state, we should model fallthroughs like how the C++ dispatcher work -- this should be on a per-dispatch key level, not a per-operator level

Reply:
If you delete 

DispatchKey.Autograd,
            DispatchKey.AutogradCPU,
            DispatchKey.AutogradCUDA,
            
            from this list then you can repro it in eager mode. That will be easier to debug.

Reply:
(you can keep prototyping, I want to refactor the python dispatcher a bit to make some more sense, including the fallback business)